consensus/src/units/store.rs

```diff
@@ -28,6 +77,44 @@ impl<'a, H: Hasher, D: Data, KB: KeyBox> UnitStore<'a, H, D, KB> {
         }
     }
 
+    pub fn get_status(&self) -> UnitStoreStatus {
+        fn first_missing(v: Vec<Round>) -> Round {
+            let mut w = vec![false; v.len()];
+            for r in &v {
+                if let Some(f) = w.get_mut(*r as usize) {
+                    *f = true;
+                }
+            }
+            match w.into_iter().position(|x| !x) {
+                Some(r) => r as u16,
+                None => v.len() as u16,
+            }
+        }
+        let n_nodes: NodeCount = self.is_forker.size().into();
+        let gm = self
+            .by_coord
+            .keys()
+            .map(|c| (c.creator, c.round))
+            .into_grouping_map();
+        let top_row: HashMap<NodeIndex, Round> = gm.clone().max();
```

Comment:
You can immediately make this a `NodeMap` using the `*_hashmap` function and this would also let you skip any type annotations here.

Reply:
Actually same the line below.

Reply:
fixed